ddtrace/internal/stats.py

```diff
@@ -0,0 +1,78 @@
+import collections
+import itertools
+import threading
+
+
+class Stats(object):
+    SPANS_STARTED = 'datadog.tracer.spans.started'
+    SPANS_FINISHED = 'datadog.tracer.spans.finished'
+    ERROR_LOGS = 'datadog.tracer.log.errors'
+    PATCH_ERROR = 'datadog.tracer.patch.error'
+    PATCH_SUCCESS = 'datadog.tracer.patch.success'
+
+    def __init__(self):
+        self._read_lock = threading.Lock()
+
+        self._last_values = collections.defaultdict(int)
+        self._values = collections.defaultdict(itertools.count)
```

Comment:
You have a good opportunity to simplify the design by having a custom `Counter` class (like the one I implemented in a previous PR comment).

Mixing both the `Counter` and the `Stats` class makes it more complicated to read, to test and maintain.

To optimize the locking, as you did here, you can pass make the `Counter` class receive optionally an already created `Lock`. That'll allow you to share the read lock across all `Counter`s.